### ISSUE TITLE
[Core] Release system

### ISSUE THREAD

**Issue by swttt** [swttt/HomeyDash#72]
I want to use this issue to discus the release system.

In general the main idea behind this is to keep track on changes, versions for now and for later releases on Cordova. 

Since beta is going to be for Patrons and closed beta testers, the release cycle is important as well. I don't want to stall beta for profit in any way, Patreon should still be seen as supporting and not buying into a closed beta. So i was thinking about weekly or bi-weekly releases to the stable branch.

For now the idea is:
- New features get added to the develop branch first
- When stable enough to go beta it would get a version bump and the commits will get added to the change log
- Matching the release cycle it would get pushed to the master branch and available for everyone

Things to discus:
- Beta only for Patrons a good idea?
- Split Patrons by pledge amount, e.g. Normal user gets access to beta widgets and plugins where Power users and up get access to beta core as well.
- Release cycle (weekly or every 2 weeks?)
- How to keep track of the change log and versioning

**Comment by lubbertkramer** [comment]
I like the idea, the question is can you keep up with a bi-weekly release and improve Homeyda.sh in that time meaningfull to don't disappoint any paying "customers", also when there are more busy times? It's no offense but good to keep in mind that it's not your main job and users could get "spoiled" just to say. Good example is the bi-weekly update from Athom wich because of delays in the software got cancelled because there was no meaning for it or it was only a repeating message with the same information. 

Paying for the beta seems like a good model to give you some revenue and to seperate the users from the "super"user which mostly also gives more and better input/github reports :)

**Comment by swttt** [comment]
I totally agree to that, the downside of the whole idea is that once it's stable i'll lose patrons (people won't see the benefit of running beta). And some day i will get to that point...

On the other hand, other devs are picking up the widget and plugin system nicely, that means the whole release cycle just isn't based on my own code.

Another important point is, that when it's stable enough, i want to port this to cordova. So it almost looks like a native app and doesn't needs the servers and domain anymore to run HomeyDash.
